### DISCUSSION TITLE
Using client-side WASM, how do I add a appsettings.json file?

### DISCUSSION THREAD

**Discussion by schnerring** [AvaloniaUI/Avalonia#12873] [Q&A] [answered]
Using the `xplat` template, I have the following projects:

- MyApp
- MyApp.Desktop
- MyApp.Browser

I added an appsettings.json to the root of `MyApp` and the following to the App.axaml.cs file of `MyApp` project:

```csharp
var configuration = new ConfigurationBuilder()
    .AddJsonFile("appsettings.json", false, true)
    .Build();
```

This works fine when starting `MyApp.Desktop`, but `MyApp.Browser` complains:

```text
Uncaught ManagedError ManagedError: One or more errors occurred. (The configuration file 'appsettings.json' was not found and is not optional. The expected physical path was '/appsettings.json'.)
```

Should I instead put two separate appsettings.json into each `MyApp.Desktop` and `MyApp.Browser` and load the config inside each Program.cs? How do I do that? For `MyApp.Browser`, I tried `wwwroot/appsettings.json` and `AppBundle/appsettings.json`, but the applicaton can't find it.

**Comment by maxkatz6** [comment, marked as answer]
Since it's a web app, you need to use HttpClient to fetch it from wwwroot. 

Alternatively, you can try to embed configuration file into assembly resources and read it with standard .net resources apis.
